Makefile_packaging.mk

```diff
@@ -289,18 +289,36 @@ debs: $(DEBS)
 ls: $(TARGETS)
 	ls -ld $^
 
+# *_LOCAL_* repos are locally built packages.
+# *_GROUP_* repos are a local mirror of a group of upstream repos.
+# *_GROUP_* repos may not supply a repomd.xml.key.
 ifneq ($(REPOSITORY_URL),)
 ifneq ($(DAOS_STACK_EL_7_LOCAL_REPO),)
-el7_LOCAL_REPOS   := $(REPOSITORY_URL)/$(DAOS_STACK_EL_7_LOCAL_REPO)/
+el7_LOCAL_REPOS   := $(REPOSITORY_URL)$(DAOS_STACK_EL_7_LOCAL_REPO)/
+endif
+SUSE_REPO_KEYS := $(OPENSUSE_MIRROR)repositories/home:/jhli/SLE_15/repodata/repomd.xml.key
+# Only needed for LEAP 42.3
+SUSE_REPO_KEYS += $(OPENSUSE_MIRROR)/repositories/science:/HPC/openSUSE_Leap_42.3/repodata/repomd.xml.key
+ifneq ($(DAOS_STACK_EL_7_GROUP_REPO),)
+el7_LOCAL_REPOS   += $(REPOSITORY_URL)$(DAOS_STACK_EL_7_GROUP_REPO)/
 endif
 ifneq ($(DAOS_STACK_SLES_12_LOCAL_REPO),)
-sle12_LOCAL_REPOS := $(REPOSITORY_URL)/$(DAOS_STACK_SLES_12_LOCAL_REPO)/
+sle12_LOCAL_REPOS := $(REPOSITORY_URL)$(DAOS_STACK_SLES_12_LOCAL_REPO)/
+endif
+ifneq ($(DAOS_STACK_SLES_12_GROUP_REPO),)
+sle12_LOCAL_REPOS += $(REPOSITORY_URL)$(DAOS_STACK_SLES_12_GROUP_REPO)/
 endif
 ifneq ($(DAOS_STACK_LEAP_42_LOCAL_REPO),)
-sl42_LOCAL_REPOS  := $(REPOSITORY_URL)/$(DAOS_STACK_LEAP_42_LOCAL_REPO)/
+sl42_LOCAL_REPOS  := $(REPOSITORY_URL)$(DAOS_STACK_LEAP_42_LOCAL_REPO)/
+endif
+ifneq ($(DAOS_STACK_LEAP_42_GROUP_REPO),)
+sl42_LOCAL_REPOS  += $(REPOSITORY_URL)$(DAOS_STACK_LEAP_42_GROUP_REPO)/
 endif
 ifneq ($(DAOS_STACK_LEAP_15_LOCAL_REPO),)
-sl15_LOCAL_REPOS  := $(REPOSITORY_URL)/$(DAOS_STACK_LEAP_15_LOCAL_REPO)/
+sl15_LOCAL_REPOS  := $(REPOSITORY_URL)$(DAOS_STACK_LEAP_15_LOCAL_REPO)/
+endif
+ifneq ($(DAOS_STACK_LEAP_15_GROUP_REPO),)
+sl15_LOCAL_REPOS  += $(REPOSITORY_URL)$(DAOS_STACK_LEAP_15_GROUP_REPO)/
```

Comment:
I was on the fence about it yesterday when I added the code I added, but I think this eleven-teen branches could be reduced down into a single set of ever-lasting/never-needing-updating by using something like:

```
DISTRO_ID_BASE := $($(basename $(DISTRO_ID))
ifneq ($(DAOS_STACK_$(DISTRO_ID_BASE)_LOCAL_REPO),)
$(DISTRO_ID_BASE)_LOCAL_REPOS  := $(REPOSITORY_URL)/$(DAOS_STACK_$(DISTRO_ID_BASE)_LOCAL_REPO)/		￼$(DISTRO_ID_BASE)_LOCAL_REPOS  := $(REPOSITORY_URL)$(DAOS_STACK_$(DISTRO_ID_BASE)_LOCAL_REPO)/
endif
￼ifneq ($(DAOS_STACK_$(DISTRO_ID_BASE)_GROUP_REPO),)
￼$(DISTRO_ID_BASE)_LOCAL_REPOS  += $(REPOSITORY_URL)$(DAOS_STACK_$(DISTRO_ID_BASE)_GROUP_REPO)/
endif
endif
```

Would of course require some realignment of global variable names, but maybe it's worth it to not have to keep writing this code for new distros.

Reply:
Reworked it to use the Distro name and major version number to lookup the environment variables for SUSE and CentOS Targets.  Should now even work for EL8 when we get to it.